idioms/str-vs-string.md

```diff
@@ -0,0 +1,90 @@
+# Using &str instead of &String
+
+## Description
+
+It is usually recommended to prefer a `&str` type over a `&String` type.  In most cases `&str` has more flexibility.  An `&String` will coerce to a `&str` type when required -- however the oppositive is not true.  
+
+## Example
+
+Let's look at a few examples.  Consider an example where we wish to determine if a word contains three consecutive vowels.  We certainly don't need to own the string to determine this so we will take a reference.  The code might look something like this:
+
+``` rust
+fn has_three_vowles(word: &String) -> bool {
+    let mut vowel_count = 0;
+    for char in word.chars() {
+        match char {
+            'a' | 'e' | 'i' | 'o' | 'u' => {
+                vowel_count += 1;
+                if vowel_count >= 3 {
+                    return true
+                }
+            }
+            _ => vowel_count = 0
+        }
+    }
+    false
+}
+```
+
+This example will work fine, as shown here:
+
+``` rust
+fn main() {
+    let ferris = "Ferris".to_string();
+    let curious = "Curious".to_string();
+    println!("{}: {}", ferris, three_vowles(&ferris));
```

Comment:
The function is defined as `has_three_vowles`, but all references use the name `three_vowles`.